controllers/cinderapi_controller.go

```diff
@@ -523,6 +523,30 @@ func (r *CinderAPIReconciler) reconcileNormal(ctx context.Context, instance *cin
 	configMapVars[ospSecret.Name] = env.SetValue(hash)
 	// run check OpenStack secret - end
 
+	//
+	// check for required TransportURL secret holding transport URL string
+	//
+	transportUrlSecret, hash, err := secret.GetSecret(ctx, helper, instance.Spec.TransportUrlSecret, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.InputReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				condition.InputReadyWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Second * 10}, fmt.Errorf("TransportURL secret %s not found", instance.Spec.TransportUrlSecret)
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.InputReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.InputReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	configMapVars[transportUrlSecret.Name] = env.SetValue(hash)
+	// run check TransportURL secret - end
```

Comment:
I don't think there is, because there's no interface for `instance.Status.Conditions`.  Every service has a different `instance` type and therefore a different `Status` type as well.

Reply:
OK, we can leave the code deduplication for a later time.